Feedbackchanges/on4 t 47 feedback

wok-gocaspi opened the pull request:
[ON4T-47]

wok-gocaspi committed:
!please ignore, dev in progress!

wok-gocaspi committed:
added GetAll Endpoint with no error handling

wok-gocaspi committed:
new GetAllEmployee Endpoint is feature complete.
added and fixed some units tests aswell

wok-gocaspi committed:
Merge branch 'master' of github.com:lucawok/example-project into getAllEmployees/ON4T-40_newEndpoint

wok-gocaspi committed:
added function "doUserExist" to check if a user with this id exists and employees can now be added by sending a non array

wok-gocaspi committed:
deleted fmt's

wok-gocaspi committed:
system now checks single jsons and json arrays if user exists on create endpoint

wok-gocaspi committed:
added unit tests

nil-gocaspi commented on handler/handler_test.go:
```diff
@@ -126,3 +128,239 @@ func TestGetAllEmployeesHandler_Return_invalid_status_code(t *testing.T) {
+	fakeService := &handlerfakes.FakeServiceInterface{}
+	var fakeEmployees []model.Employee
+	var fakeEmployee model.Employee
+	fakeEmployee.ID = "100"
```
Es ist schöner die Felder direkt im Objekt anzugeben, wie du das auch in den folgenden Tests gemacht hast.

nil-gocaspi commented on handler/handler_test.go:
```diff
@@ -126,3 +128,239 @@ func TestGetAllEmployeesHandler_Return_invalid_status_code(t *testing.T) {
+	fakeEmployee.LastName = "Tester"
+	fakeEmployee.FirstName = "Test"
+	fakeEmployee.Email = "tester@example.com"
+	fakeEmployees = append(fakeEmployees, fakeEmployee)
```
Die kannst du ebenfalls sparen wenn du fakeEmployees direkt im Objekt initalisiertst :
fakeEmployees := []model.Employee{
model.Employee{id:1,last_name: ....}
}

nil-gocaspi commented on handler/handler_test.go:
```diff
@@ -126,3 +128,239 @@ func TestGetAllEmployeesHandler_Return_invalid_status_code(t *testing.T) {
+	var fakeEmployees []model.Employee
+	var fakeEmployee model.Employee
+	fakeEmployee.ID = "100"
+	fakeEmployee.LastName = "Tester"
```
siehe oben

nil-gocaspi commented:
The createEmployee handler has a logical error, if I create more than 1 Employee, whereas some of them have the same Id, than the unique check doesnt throw an error

wok-gocaspi committed:
NIL Verify Edits
-changed doUserExist Handler to be able to detect if there are identical user ids within a json array and added/changed units tests to cover changes
-simplified employee model variables within the handler unit tests

wok-gocaspi committed:
NIL Verify Edits 2
- improved TestDoUserExist_false unit test to simulate if two diffrent users are given instead of using an empty slice

nil-gocaspi commented:
OK!
